cargo-dist/tests/gallery/dist.rs

```diff
@@ -366,6 +369,221 @@ impl DistResult {
+            let app_home = tempdir.join(format!(".{app_name}"));
+            let _output = powershell.output_checked(|cmd| {
+                cmd.arg("-c")
+                    .arg(shell_path)
+                    .env("UserProfile", &tempdir)
+                    .env("LOCALAPPDATA", &appdata)
+                    .env("MY_ENV_VAR", &app_home)
+                    .env_remove("CARGO_HOME")
+                    .env_remove("PSModulePath")
+            })?;
+
+            // Check that the script wrote files where we expected
+            let receipt_file = appdata.join(format!("{app_name}\\{app_name}-receipt.json"));
+            let expected_bin_dir = Utf8PathBuf::from(expected_bin_dir.replace('/', "\\"));
+            let bin_dir = tempdir.join(&expected_bin_dir);
+
+            assert!(bin_dir.exists(), "bin dir wasn't created");
+
+            // Check that all the binaries work
+            for bin_name in ctx.repo.bins {
+                let bin_path = bin_dir.join(format!("{bin_name}.exe"));
+                assert!(bin_path.exists(), "bin wasn't created");
+
+                let bin =
+                    CommandInfo::new(bin_name, Some(bin_path.as_str())).expect("failed to run bin");
+                assert!(bin.version().is_some(), "failed to get app version");
+                eprintln!("installer.ps1 worked!");
+
+                // checking path...
+                // Make a test.ps1 script that runs `where.exe {bin_name}`
+                //
+                // (note that "where" and "where.exe" are completely different things...)
+                //
+                // also note that HKCU:\Environment\PATH is not actually the full PATH
+                // a shell will have, so preprend it to the current PATH (if we don't do
+                // this then where.exe won't be on PATH anymore!)
+                let empirical_path = run_ps1_script(
+                    &powershell,
+                    &tempdir,
+                    "test.ps1",
+                    &format!(
+                        r#"
+                $Item = Get-Item -Path "HKCU:\Environment"
+                $RegPath = $Item | Get-ItemPropertyValue -Name "Path"
+                $env:PATH = "$RegPath;$env:PATH"
+                $Res = where.exe {bin_name}
+                return $Res
+                "#
+                    ),
+                )?;
+                // where.exe will return every matching result, but the one we
+                // want, the one selected by PATH, should appear first.
+                assert_eq!(
+                    empirical_path.lines().next().unwrap(),
+                    bin_path.as_str(),
+                    "bin path wasn't right"
+                );
+            }
+
+            // check the install receipts
+            // FIXME: temporarily disabled for the feature being broken
+            // self.check_install_receipt(ctx, &bin_dir, &receipt_file, ".exe");
```

Comment:
todo for @mistydemeo: fix install receipts and uncomment this line

Reply:
Rebase with this removed works: https://github.com/axodotdev/cargo-dist/pull/866

Reply:
You can reenable this now that #863 is merged.